app/routes/list-project.tsx

```diff
@@ -0,0 +1,531 @@
+import cx from "classnames";
+import Button from "~/components/Button";
+import TextArea from "~/components/TextArea";
+import TextField from "~/components/TextField";
+import {
+  ROLE_INTERESTS,
+  SUBJECT_INTERESTS,
+  TECH_INTERESTS,
+} from "~/utils/tags";
+import formStyles from "~/styles/forms.css";
+import markdownStyles from "~/styles/markdown.css";
+import TwitterIcon from "~/components/icons/TwitterIcon";
+import { LinkIcon, MarkGithubIcon } from "@primer/octicons-react";
+import ProjectPreview from "~/components/ProjectPreview";
+import { Project } from "~/types";
+import ProjectSubmissionSpinner from "~/components/ProjectSubmissionSpinner";
+import RequiredMarker from "~/components/RequiredMarker";
+import ExternalLink from "~/components/ExternalLink";
+import mapThumbnailSrc from "~/images/map_thumbnail.png";
+import { formatFileSize, maybeStringToArray } from "~/utils/formatting";
+import FieldError from "~/components/FieldError";
+import {
+  ActionFunction,
+  json,
+  LoaderFunction,
+  redirect,
+} from "@remix-run/node";
+import {
+  getAccessToken,
+  getCurrentUser,
+  getCurrentUserOrRedirect,
+} from "~/session.server";
+import {
+  MAX_AVATAR_SIZE_BYTES,
+  parseListProjectForm,
+} from "~/utils/project-submission";
+import { createNewPullRequest } from "~/github.server";
+import { getRepoOwnerAndName } from "~/utils/repo-url";
+
+export function links() {
+  return [
+    { rel: "stylesheet", href: formStyles },
+    { rel: "stylesheet", href: markdownStyles },
+  ];
+}
+
+/**
+ * Check that the user is logged in or redirect to the /login page
+ */
+export const loader: LoaderFunction = async ({ request }) => {
+  await getCurrentUserOrRedirect(request);
+  return json({});
+};
+
+/**
+ * This method is called when the form is submitted. It performs data validation
+ * and uploads files to GitHub before creating a PR. If all goes well, it
+ * redirects to a success page.
+ */
+export const action: ActionFunction = async ({ request }) => {
+  const currentUser = await getCurrentUser(request);
+  const accessToken = await getAccessToken(request);
+
+  if (!currentUser || !accessToken) {
+    throw new Error("You must be logged in to save your profile");
+  }
+
+  // TODO check whether we already have a project for this repository
+
+  // Parse the form
+  const { validationErrors, files, repoUrl } = await parseListProjectForm(
+    request,
+    currentUser
+  );
+
+  // If there are any validation errors, return early
+  if (validationErrors) {
+    return json({ validationErrors });
+  }
+
+  if (!repoUrl) {
+    throw new Error("Missing repo URL in new project form");
+  }
+
+  // Create a new pull request with the content of the form
+  const { owner: repoOwner, name: repoName } = getRepoOwnerAndName(repoUrl);
+  const { pullRequestUrl } = await createNewPullRequest(
+    accessToken,
+    files,
+    repoOwner,
+    repoName
+  );
+
+  return redirect(`/project-listed?pr=${encodeURIComponent(pullRequestUrl)}`);
```

Comment:
And when eeeeeverything is complete, we redirect the user to a new page and show them a link to their fresh new PR.